Automattic/newspack-newsletters
# feat: ads taxonomies and fields

### All Submissions:






### Changes proposed in this Pull Request:

<img width="1318" alt="image" src="https://github.com/Automattic/newspack-newsletters/assets/820752/25ab0c47-207e-4b46-9d4d-1e1ce0b0cbf1">

Implements the following new tax and meta fields for Newsletter Ads:
 - Start date field
 - Price field
 - Advertisers taxonomy
 - Categories taxonomy (same as posts/newsletters)

The taxonomies determine the following:
1. Newsletters associated with an advertiser should only render ads that are also associated with that same advertiser
2. Ads associated with a category should only be rendered in newsletters that are also associated with that category

That means newsletters without any advertiser association can render ads from any advertiser. That also means that ads without any categories can be rendered on any newsletter.

### How to test the changes in this Pull Request:

1. Check out this branch and create an ad with a future start date
2. Draft a new newsletter, save, click to preview, and confirm the ad doesn't render
3. Edit the ad and toggle "Custom Start Date"
4. Don't click any date, the starting date is today and should be disabled for selection (just like past dates)
5. Save, edit the newsletter, and preview
6. Confirm the ad renders
7. Edit the ad, set a future start date, and toggle the "Expiration Date"
8. Confirm you can't select a date prior to the start date
9. Select a future date, set a price, and save
10. Refresh and confirm the values persisted
11. Untoggle the Custom Start Date and save
12. Edit the newsletter, save, preview, and confirm you the ad renders
13. Edit the ad and attach it to an existing category and a new advertiser
14. Edit the newsletter, save, preview, and confirm the ad doesn't render
15. Set the category to the newsletter, save, preview, and confirm the ad renders
16. Set a different new advertiser, save, preview, and confirm the ad doesn't render

### Other information:
